jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHandler.java

```diff
@@ -434,41 +437,42 @@ public void doScope(String target, Request baseRequest, HttpServletRequest reque
         // Get the base requests
         final String old_servlet_path = baseRequest.getServletPath();
         final String old_path_info = baseRequest.getPathInfo();
-        final PathSpec old_path_spec = baseRequest.getPathSpec();
+        final HttpServletMapping old_http_servlet_mapping = baseRequest.getHttpServletMapping();
 
         DispatcherType type = baseRequest.getDispatcherType();
 
         ServletHolder servletHolder = null;
         UserIdentity.Scope oldScope = null;
 
-        MappedResource<ServletHolder> mapping = getMappedServlet(target);
+        MappedServlet mapping = getMappedServlet(target);
         if (mapping != null)
         {
             servletHolder = mapping.getResource();
 
-            if (mapping.getPathSpec() != null)
+            PathSpec pathSpec = mapping.getPathSpec();
+            if (pathSpec != null)
             {
-                PathSpec pathSpec = mapping.getPathSpec();
                 String servletPath = pathSpec.getPathMatch(target);
```

Comment:
why are we still getting serlvetPath and pathInfo directly from the pathSpec.   I feel like we can do this more efficiently if we asked the mapping, as it will already know if it is an exact match etc.